src/Microsoft.Tye.Core/Serialization/YamlIngressHelpers.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.Tye.ConfigModel;
+using YamlDotNet.RepresentationModel;
+
+namespace Tye.Serialization
+{
+    public static class YamlIngressHelpers
+    {
+        public static void HandleIngress(YamlSequenceNode yamlSequenceNode, List<ConfigIngress> ingress)
+        {
+            foreach (var child in yamlSequenceNode.Children)
+            {
+                switch (child.NodeType)
+                {
+                    case YamlNodeType.Mapping:
+                        var configIngress = new ConfigIngress();
+                        HandleIngressMapping(child as YamlMappingNode, configIngress);
+                        ingress.Add(configIngress);
+                        break;
+                    default:
+                        throw new TyeYamlException(child.Start,
+                            CoreStrings.FormatUnexpectedType(YamlNodeType.Mapping.ToString(), child.NodeType.ToString()));
+                }
+            }
+        }
+
+        private static void HandleIngressMapping(YamlMappingNode? yamlMappingNode, ConfigIngress configIngress)
+        {
+            foreach (var child in yamlMappingNode!.Children)
```

Comment:
nullability tip. If you find you find yourself using defining a nullable parameter and then immediately asserting it's non-null then the parameter should not be nullable. The caller needs to do the checking.

Reply:
I think this is just inconsistent. I have other places where I removed the nullability check. I need to do a second pass of this when I have a fresh mind as there is some low hanging fruit.